lib/rex/proto/dns/server.rb

```diff
@@ -293,14 +164,14 @@ def default_dispatch_request(cli,data)
       if cached.empty?
         next
       else
-        req.answer = req.answer + cached
+        req.instance_variable_set(:@answer, (req.answer + cached).uniq)
         forward.question.delete(ques)
       end
     end
     # Forward remaining requests, cache responses
     if forward.question.count > 0 and @fwd_res
-      forwarded = self.fwd_res.send(validate_packet(forward))
-      req.answer = req.answer + forwarded.answer
+      forwarded = self.fwd_res.send(forward)
+      req.instance_variable_set(:@answer, (req.answer + forwarded.answer).uniq)
```

Comment:
Is there a reason this is used vs setting the value of req.answer? As noted in the docs this "Sets the instance variable named by symbol to the given object, thereby frustrating the efforts of the class’s author to attempt to provide proper encapsulation" which doesn't sound to me like an ideal solution 😄

Reply:
IIRC its a hack to deal with the various packet types people put into their DNS code. I still see modules getting submitted using `Net::DNS`